Bug in Either `map_over_sweeps` or in `to_cfradial1`

* xradar version: 0.8.1
* Python version: 3.12.7
* Operating System: MacOS

### Description
When applying any operation on a datatree using `map_over_sweeps`, it produces an error, although they look identical when compared to each other. 


### What I Did

```python
import cmweather
import numpy as np
import xarray as xr
import xradar as xd
from open_radar_data import DATASETS
xd.__version__
```
```
'0.8.1.dev7+g830d86b'
```

```python
file = DATASETS.fetch("swx_20120520_0641.nc")

dtree = xd.io.open_cfradial1_datatree(file)

xd.io.to_cfradial1(dtree, filename='test_cfradial1.nc')

dtree2 = xd.io.open_cfradial1_datatree("test_cfradial1.nc")

np.testing.assert_equal(dtree.groups, dtree2.groups)

np.testing.assert_equal(dtree, dtree2)

xd.io.to_cfradial1(dtree2, filename='test_cfradial2.nc')

dtree3 = dtree.xradar.georeference()
dtree3['sweep_0']['corrected_reflectivity_horizontal'].plot(x='x', y='y', vmin=-10, vmax=75, cmap='NWSRef')
```
![image](https://github.com/user-attachments/assets/62ae6136-c9b3-499e-aa8d-4bf632a4f21a)

```python
def filter_radar(ds, vel_name='VELH', ref_name='DBZH'):
    vel_texture = ds[vel_name].rolling(range=30, min_periods=2, center=True).std()
    ds = ds.assign(velocity_texture=vel_texture)
    ds = ds.where((ds.velocity_texture<10) & ((ds[ref_name]>=-10) & (ds[ref_name]<=75)))
    return ds

dtree3 = dtree3.xradar.map_over_sweeps(filter_radar,
                                       vel_name='mean_doppler_velocity',
                                       ref_name='corrected_reflectivity_horizontal')

dtree3['sweep_0']['corrected_reflectivity_horizontal'].plot(x='x', y='y', vmin=-10, vmax=75, cmap='NWSRef')
```

![image](https://github.com/user-attachments/assets/5d50f2bc-bfd1-4fde-b131-900a55983624)

```python
xd.io.to_cfradial1(dtree3, 'test_cfradial_filt.nc')
```

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[37], line 1
----> 1 xd.io.to_cfradial1(dtree3, 'test_cfradial_filt.nc')

File ~/micromamba/envs/xradar_latest/lib/python3.12/site-packages/xradar/io/export/cfradial1.py:298, in to_cfradial1(dtree, filename, calibs)
    283 """
    284 Convert a radar xarray.DataTree to the CFRadial1 format
    285 and save it to a file. Ensure that the resulting dataset
   (...)
    295     Whether to include calibration parameters.
    296 """
    297 # Generate the initial ds_cf using the existing mapping functions
--> 298 dataset = _variable_mapper(dtree)
    300 # Handle calibration parameters
    301 if calibs:

File ~/micromamba/envs/xradar_latest/lib/python3.12/site-packages/xradar/io/export/cfradial1.py:105, in _variable_mapper(dtree, dim0)
     88 def _variable_mapper(dtree, dim0=None):
     89     """
     90     Map radar variables for different sweep groups.
     91 
   (...)
    102         Dataset containing mapped radar variables.
    103     """
--> 105     sweep_info = _sweep_info_mapper(dtree)
    106     vol_info = _main_info_mapper(dtree)
    107     if "fixed_angle" in vol_info:

File ~/micromamba/envs/xradar_latest/lib/python3.12/site-packages/xradar/io/export/cfradial1.py:198, in _sweep_info_mapper(dtree)
    184 sweep_vars = [
    185     "sweep_number",
    186     "sweep_mode",
   (...)
    192     "sweep_end_ray_index",
    193 ]
    195 for var_name in sweep_vars:
    196     var_data_list = [
    197         (
--> 198             np.unique(dtree[s][var_name].values[np.newaxis, ...])
    199             if var_name in dtree[s]
    200             else np.array([np.nan])
    201         )
    202         for s in dtree.groups
    203         if "sweep" in s
    204     ]
    206     var_attrs_list = [
    207         dtree[s][var_name].attrs if var_name in dtree[s] else {}
    208         for s in dtree.groups
    209         if "sweep" in s
    210     ]
    212     if not var_data_list:

File ~/micromamba/envs/xradar_latest/lib/python3.12/site-packages/numpy/lib/_arraysetops_impl.py:289, in unique(ar, return_index, return_inverse, return_counts, axis, equal_nan)
    287 ar = np.asanyarray(ar)
    288 if axis is None:
--> 289     ret = _unique1d(ar, return_index, return_inverse, return_counts, 
    290                     equal_nan=equal_nan, inverse_shape=ar.shape, axis=None)
    291     return _unpack_tuple(ret)
    293 # axis was specified and not None

File ~/micromamba/envs/xradar_latest/lib/python3.12/site-packages/numpy/lib/_arraysetops_impl.py:356, in _unique1d(ar, return_index, return_inverse, return_counts, equal_nan, inverse_shape, axis)
    354     aux = ar[perm]
    355 else:
--> 356     ar.sort()
    357     aux = ar
    358 mask = np.empty(aux.shape, dtype=np.bool)

TypeError: '<' not supported between instances of 'float' and 'str'
```

```python
xr.testing.assert_identical(dtree['sweep_0'], dtree3['sweep_0'])
```

```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[43], line 1
----> 1 xr.testing.assert_identical(dtree['sweep_0'], dtree3['sweep_0'])

    [... skipping hidden 1 frame]

File ~/micromamba/envs/xradar_latest/lib/python3.12/site-packages/xarray/testing/assertions.py:181, in assert_identical(a, b)
    179     assert a.identical(b), formatting.diff_coords_repr(a, b, "identical")
    180 elif isinstance(a, DataTree):
--> 181     assert a.identical(b), diff_datatree_repr(a, b, "identical")
    182 else:
    183     raise TypeError(f"{type(a)} not supported by assertion comparison")

AssertionError: Left and right DataTree objects are not identical

Data at root node does not match:
    Differing data variables:
    L   diff_reflectivity                  (azimuth, range) float32 1MB 0.0 ... 0.0
    R   diff_reflectivity                  (azimuth, range) float32 1MB 0.0 ... nan
    L   recalculated_diff_phase            (azimuth, range) float32 1MB -0.0 ... ...
    R   recalculated_diff_phase            (azimuth, range) float32 1MB -0.0 ... nan
    L   diff_phase                         (azimuth, range) float32 1MB 0.0 ... nan
    R   diff_phase                         (azimuth, range) float32 1MB 0.0 ... nan
    L   proc_dp_phase_shift                (azimuth, range) float32 1MB 18.09 ......
    R   proc_dp_phase_shift                (azimuth, range) float32 1MB 18.09 ......
    L   corrected_reflectivity_horizontal  (azimuth, range) float32 1MB -5.617 .....
    R   corrected_reflectivity_horizontal  (azimuth, range) float32 1MB -5.617 .....
    L   specific_attenuation               (azimuth, range) float32 1MB 0.0004272...
    R   specific_attenuation               (azimuth, range) float32 1MB 0.0004272...
    L   mean_doppler_velocity              (azimuth, range) float32 1MB 0.0 ... -...
    R   mean_doppler_velocity              (azimuth, range) float32 1MB 0.0 ... nan
    L   rain_rate_A                        (azimuth, range) float32 1MB 0.0625 .....
    R   rain_rate_A                        (azimuth, range) float32 1MB 0.0625 .....
    L   norm_coherent_power                (azimuth, range) float32 1MB 1.0 ... 0.17
    R   norm_coherent_power                (azimuth, range) float32 1MB 1.0 ... nan
    L   copol_coeff                        (azimuth, range) float32 1MB 1.0 ... nan
    R   copol_coeff                        (azimuth, range) float32 1MB 1.0 ... nan
    L   sweep_mode                         <U20 80B 'azimuth_surveillance'
    R   sweep_mode                         (azimuth, range) object 2MB 'azimuth_s...
    L   sweep_fixed_angle                  float64 8B 0.5
    R   sweep_fixed_angle                  (azimuth, range) float64 2MB 0.5 ... nan
    L   sweep_number                       int32 4B 0
    R   sweep_number                       (azimuth, range) float64 2MB 0.0 ... nan
    L   unf_dp_phase_shift                 (azimuth, range) float32 1MB 0.0 ... 28.4
    R   unf_dp_phase_shift                 (azimuth, range) float32 1MB 0.0 ... nan
    L   dp_phase_shift                     (azimuth, range) float32 1MB 90.0 ... nan
    R   dp_phase_shift                     (azimuth, range) float32 1MB 90.0 ... nan
    L   reflectivity_horizontal            (azimuth, range) float32 1MB -6.203 .....
    R   reflectivity_horizontal            (azimuth, range) float32 1MB -6.203 .....
    Data variables only on the right object:
        velocity_texture                   (azimuth, range) float32 1MB 4.744 ......

```

I tried it on another dataset, and even though they look identical, they still can’t be exported or transformed to `cfradial1` after applying any operation using `map_over_sweeps`

![temp_compare_dtree](https://github.com/user-attachments/assets/ff7b976c-798a-4ce7-8f6a-df03d0eb1569)

--- a/docs/history.md
+++ b/docs/history.md
@@ -6,6 +6,9 @@
 
 ## 0.12.0 (2026-04-21)
 
+## Development
+
+* FIX: Preserve CfRadial metadata groups during CfRadial1/CfRadial2 transforms, allow ``to_cfradial2`` to reopen root-only ``engine="cfradial1"`` datasets, and normalize sweep metadata after ``map_over_sweeps`` operations so filtered volumes can be exported to CfRadial1 again ({issue}`254`, {issue}`322`) by [@syedhamidali](https://github.com/syedhamidali)
 * MNT: Unpin xarray, require ``xarray >= 2026.4.0`` in ``requirements.txt``, ``environment.yml``, ``ci/unittests.yml``, and ``ci/notebooktests.yml`` by [@aladinor](https://github.com/aladinor)
 * MNT: Clarify contributor, team-member, and maintainer roles in the contributing guide, including the pathway to greater project involvement ({issue}`341`) by [@kmuehlbauer](https://github.com/kmuehlbauer), ({pull}`354`) by [@syedhamidali](https://github.com/syedhamidali)
 * FIX: ``open_nexradlevel2_datatree`` keeps sweeps with interior sweep-index gaps — derive sweep names from actual indices in ``nex.data`` instead of ``range(len(...))`` so upstream-dropped interior cuts (e.g. ``[0..9, 11]``) no longer raise ``KeyError`` ({issue}`361`, {pull}`362`) by [@aladinor](https://github.com/aladinor)

--- a/docs/history.md
+++ b/docs/history.md
@@ -2,13 +2,11 @@
 
 ## Development
 
+* FIX: Preserve CfRadial metadata groups during CfRadial1/CfRadial2 transforms, allow ``to_cfradial2`` to reopen root-only ``engine="cfradial1"`` datasets, and normalize sweep metadata after ``map_over_sweeps`` operations so filtered volumes can be exported to CfRadial1 again ({issue}`254`, {issue}`322`) by [@syedhamidali](https://github.com/syedhamidali)
 * ENH: Expose ``target_crs`` on ``.xradar.georeference()`` accessor for DataArray, Dataset and DataTree, enabling reprojection via e.g. ``radar.xradar.georeference(target_crs=4326)`` ({issue}`243`) by [@syedhamidali](https://github.com/syedhamidali)
 
 ## 0.12.0 (2026-04-21)
 
-## Development
-
-* FIX: Preserve CfRadial metadata groups during CfRadial1/CfRadial2 transforms, allow ``to_cfradial2`` to reopen root-only ``engine="cfradial1"`` datasets, and normalize sweep metadata after ``map_over_sweeps`` operations so filtered volumes can be exported to CfRadial1 again ({issue}`254`, {issue}`322`) by [@syedhamidali](https://github.com/syedhamidali)
 * MNT: Unpin xarray, require ``xarray >= 2026.4.0`` in ``requirements.txt``, ``environment.yml``, ``ci/unittests.yml``, and ``ci/notebooktests.yml`` by [@aladinor](https://github.com/aladinor)
 * MNT: Clarify contributor, team-member, and maintainer roles in the contributing guide, including the pathway to greater project involvement ({issue}`341`) by [@kmuehlbauer](https://github.com/kmuehlbauer), ({pull}`354`) by [@syedhamidali](https://github.com/syedhamidali)
 * FIX: ``open_nexradlevel2_datatree`` keeps sweeps with interior sweep-index gaps — derive sweep names from actual indices in ``nex.data`` instead of ``range(len(...))`` so upstream-dropped interior cuts (e.g. ``[0..9, 11]``) no longer raise ``KeyError`` ({issue}`361`, {pull}`362`) by [@aladinor](https://github.com/aladinor)

--- a/tests/io/test_cfradial1.py
+++ b/tests/io/test_cfradial1.py
@@ -3,10 +3,12 @@
 # Distributed under the MIT License. See LICENSE for more info.
 
 
+import numpy as np
 import xarray as xr
 from open_radar_data import DATASETS
 
 import xradar as xd
+from xradar.io.export import cfradial1 as cf1_export
 
 
 def test_cfradial1_open_mfdataset_context_manager(cfradial1_file):
@@ -47,3 +49,99 @@ def test_compare_sweeps(temp_file):
         xr.testing.assert_equal(
             dtree[f"sweep_{sweep_num}"].ds, dtree1[f"sweep_{sweep_num}"].ds
         )
+
+
+def test_cfradial1_export_helper_scalar_normalization():
+    assert cf1_export._first_valid_scalar(xr.DataArray(np.array([np.nan, 3.0]))) == 3.0
+
+    masked = xr.DataArray(np.ma.array([1.0, 2.0], mask=[True, False]))
+    assert cf1_export._first_valid_scalar(masked) == 2.0
+
+    nat = xr.DataArray(np.array(["NaT", "2025-01-01T00:00:00"], dtype="datetime64[ns]"))
+    assert np.datetime64(cf1_export._first_valid_scalar(nat), "ns") == np.datetime64(
+        "2025-01-01T00:00:00", "ns"
+    )
+
+    text = xr.DataArray(np.array(["azimuth_surveillance"], dtype=object))
+    assert cf1_export._first_valid_scalar(text) == "azimuth_surveillance"
+
+    missing = xr.DataArray(np.array([np.nan, np.nan]))
+    assert np.isnan(cf1_export._first_valid_scalar(missing))
+
+
+def test_cfradial1_export_helper_metadata_and_indices():
+    sweep = xr.Dataset(
+        data_vars={
+            "sweep_number": (
+                ("azimuth", "range"),
+                np.array([[0.0, np.nan], [0.0, np.nan]]),
+            ),
+            "sweep_mode": (
+                ("azimuth", "range"),
+                np.array(
+                    [
+                        ["azimuth_surveillance", None],
+                        ["azimuth_surveillance", None],
+                    ],
+                    dtype=object,
+                ),
+            ),
+            "DBZ": (("azimuth", "range"), np.ones((2, 2), dtype="float32")),
+        },
+        coords={
+            "azimuth": ("azimuth", np.array([0.0, 1.0], dtype="float32")),
+            "range": ("range", np.array([100.0, 200.0], dtype="float32")),
+            "time": (
+                "azimuth",
+                np.array(["2025-01-01", "2025-01-01"], dtype="datetime64[ns]"),
+            ),
+            "elevation": ("azimuth", np.array([0.5, 0.5], dtype="float32")),
+        },
+    )
+
+    normalized = cf1_export._normalize_sweep_metadata(sweep)
+    assert normalized["sweep_number"].dims == ()
+    assert normalized["sweep_number"].item() == 0.0
+    assert normalized["sweep_mode"].dims == ()
+    assert normalized["sweep_mode"].item() == "azimuth_surveillance"
+
+    valid = xr.DataTree.from_dict(
+        {
+            "/": xr.Dataset(),
+            "/sweep_0": xr.Dataset(
+                coords={"elevation": ("azimuth", np.array([0.5, 0.5], dtype="float32"))}
+            ),
+        }
+    )
+    out = cf1_export.calculate_sweep_indices(valid)
+    assert out["sweep_start_ray_index"].dims == ("sweep",)
+    assert out["sweep_end_ray_index"].dims == ("sweep",)
+
+
+def test_cfradial1_export_helper_empty_sweep_info_and_time_fallback():
+    empty = xr.DataTree.from_dict({"/": xr.Dataset()})
+    sweep_info = cf1_export._sweep_info_mapper(empty)
+    assert "sweep_number" in sweep_info
+    assert np.isnan(sweep_info["sweep_number"].values[0])
+
+    sweep = xr.Dataset(
+        data_vars={
+            "DBZ": (("time", "range"), np.ones((2, 2), dtype="float32")),
+            "sweep_mode": ((), "manual"),
+            "sweep_number": ((), 0),
+            "sweep_fixed_angle": ((), 0.5),
+        },
+        coords={
+            "time": (
+                "time",
+                np.array(["2025-01-01", "2025-01-01T00:00:01"], dtype="datetime64[ns]"),
+            ),
+            "range": ("range", np.array([100.0, 200.0], dtype="float32")),
+            "azimuth": ("time", np.array([0.0, 1.0], dtype="float32")),
+            "elevation": ("time", np.array([0.5, 0.5], dtype="float32")),
+        },
+    )
+    dtree = xr.DataTree.from_dict({"/": xr.Dataset(), "/sweep_0": sweep})
+    mapped = cf1_export._variable_mapper(dtree)
+    assert "DBZ" in mapped
+    assert mapped["DBZ"].dims == ("time", "range")

--- a/tests/transform/test_cfradial.py
+++ b/tests/transform/test_cfradial.py
@@ -4,6 +4,7 @@
 
 import pytest
 import xarray as xr
+from open_radar_data import DATASETS
 from xarray import MergeError
 
 import xradar as xd
@@ -12,7 +13,6 @@ import xradar as xd
 def test_to_cfradial1(cfradial1_file):
     """Test the conversion from DataTree to CfRadial1 format."""
     with xd.io.open_cfradial1_datatree(cfradial1_file) as dtree:
-
         # Call the conversion function
         ds_cf1 = xd.transform.to_cfradial1(dtree)
 
@@ -30,7 +30,6 @@ def test_to_cfradial1(cfradial1_file):
 def test_to_cfradial2(cfradial1_file):
     """Test the conversion from CfRadial1 to CfRadial2 DataTree format."""
     with xd.io.open_cfradial1_datatree(cfradial1_file, optional_groups=True) as dtree:
-
         # Convert to CfRadial1 dataset first
         ds_cf1 = xd.transform.to_cfradial1(dtree)
 
@@ -40,10 +39,31 @@ def test_to_cfradial2(cfradial1_file):
         # Verify key attributes and data structures in the resulting datatree
         assert isinstance(dtree_cf2, xr.DataTree), "Output is not a valid DataTree"
         assert "radar_parameters" in dtree_cf2, "Missing radar_parameters in DataTree"
+        assert (
+            "georeferencing_correction" in dtree_cf2
+        ), "Missing georeferencing_correction in DataTree"
         # Round-tripped attrs should be a subset of the original — backend-specific
         # attrs (e.g. NEXRAD ICD metadata) are not preserved by the generic reader.
         for key, val in dtree_cf2.attrs.items():
             assert ds_cf1.attrs[key] == val, f"Attr {key!r} mismatch after round-trip"
+
+
+def test_to_cfradial2_root_only_dataset_reopens_source(cfradial1_file):
+    ds = xr.open_dataset(cfradial1_file, engine="cfradial1")
+
+    dtree = xd.transform.to_cfradial2(ds)
+
+    assert isinstance(dtree, xr.DataTree)
+    assert "georeferencing_correction" in dtree
+    assert "sweep_0" in dtree
+
+
+def test_to_cfradial2_root_only_dataset_without_source_errors(cfradial1_file):
+    ds = xr.open_dataset(cfradial1_file, engine="cfradial1").copy()
+    ds.encoding = {}
+
+    with pytest.raises(ValueError, match="does not retain a readable"):
+        xd.transform.to_cfradial2(ds)
 
 
 def test_to_cfradial1_with_different_range_shapes(nexradlevel2_bzfile):
@@ -76,3 +96,31 @@ def test_to_cfradial1_error_with_different_range_bin_sizes(gamic_file):
     with xd.io.open_gamic_datatree(gamic_file) as dtree:
         with pytest.raises(MergeError):
             xd.transform.to_cfradial1(dtree)
+
+
+def test_to_cfradial1_after_map_over_sweeps_where(tmp_path):
+    file = DATASETS.fetch("swx_20120520_0641.nc")
+    dtree = xd.io.open_cfradial1_datatree(file)
+
+    def filter_radar(
+        ds,
+        vel_name="mean_doppler_velocity",
+        ref_name="corrected_reflectivity_horizontal",
+    ):
+        vel_texture = ds[vel_name].rolling(range=30, min_periods=2, center=True).std()
+        ds = ds.assign(velocity_texture=vel_texture)
+        return ds.where(
+            (ds.velocity_texture < 10) & ((ds[ref_name] >= -10) & (ds[ref_name] <= 75))
+        )
+
+    filtered = dtree.xradar.map_over_sweeps(filter_radar)
+    ds_cf1 = xd.transform.to_cfradial1(filtered)
+
+    assert ds_cf1["sweep_number"].dims == ("sweep",)
+    assert ds_cf1["fixed_angle"].dims == ("sweep",)
+    assert ds_cf1["sweep_mode"].dims == ("sweep",)
+
+    outfile = tmp_path / "filtered_cfradial1.nc"
+    xd.io.to_cfradial1(filtered, outfile)
+    reopened = xd.io.open_cfradial1_datatree(outfile)
+    assert "sweep_0" in reopened

--- a/xradar/io/export/cfradial1.py
+++ b/xradar/io/export/cfradial1.py
@@ -33,6 +33,53 @@ import numpy as np
 import xarray as xr
 
 
+def _first_valid_scalar(data_array):
+    """Collapse a metadata variable to its first non-missing scalar value."""
+    values = np.asarray(data_array.values).ravel()
+    for value in values:
+        try:
+            if np.ma.is_masked(value):
+                continue
+            if np.isnan(value):
+                continue
+        except TypeError:
+            pass
+        try:
+            if np.isnat(value):
+                continue
+        except TypeError:
+            pass
+        return np.asarray(value).item()
+    return np.nan
+
+
+def _normalize_sweep_metadata(data):
+    """Restore sweep metadata variables to scalar form before CfRadial1 export."""
+    metadata_vars = [
+        "sweep_number",
+        "sweep_mode",
+        "polarization_mode",
+        "prt_mode",
+        "follow_mode",
+        "sweep_fixed_angle",
+        "sweep_start_ray_index",
+        "sweep_end_ray_index",
+    ]
+
+    data = data.copy()
+    for name in metadata_vars:
+        if name not in data:
+            continue
+        if data[name].ndim == 0:
+            continue
+        data[name] = xr.DataArray(
+            _first_valid_scalar(data[name]),
+            attrs=data[name].attrs,
+        )
+
+    return data
+
+
 def _calib_mapper(calib_params):
     """
     Map calibration parameters to a new dataset format.
@@ -109,13 +156,15 @@ def _variable_mapper(dtree, dim0=None):
     sweep_datasets = []
     for grp in dtree.groups:
         if "sweep" in grp:
-            data = dtree[grp].to_dataset(inherit="all_coords")
+            data = _normalize_sweep_metadata(
+                dtree[grp].to_dataset(inherit="all_coords")
+            )
 
             # handling first dimension
             if dim0 is None:
                 dim0 = (
                     "elevation"
-                    if data.sweep_mode.load().astype(str) == "rhi"
+                    if str(data.sweep_mode.load().values) == "rhi"
                     else "azimuth"
                 )
                 if dim0 not in data.dims:
@@ -200,7 +249,7 @@ def _sweep_info_mapper(dtree):
     for var_name in sweep_vars:
         var_data_list = [
             (
-                np.unique(dtree[s][var_name].values[np.newaxis, ...])
+                np.asarray([_first_valid_scalar(dtree[s][var_name])])
                 if var_name in dtree[s]
                 else np.array([np.nan])
             )

--- a/xradar/io/export/cfradial1.py
+++ b/xradar/io/export/cfradial1.py
@@ -35,21 +35,15 @@ import xarray as xr
 
 def _first_valid_scalar(data_array):
     """Collapse a metadata variable to its first non-missing scalar value."""
-    values = np.asarray(data_array.values).ravel()
-    for value in values:
-        try:
-            if np.ma.is_masked(value):
-                continue
-            if np.isnan(value):
-                continue
-        except TypeError:
-            pass
-        try:
-            if np.isnat(value):
-                continue
-        except TypeError:
-            pass
-        return np.asarray(value).item()
+    if data_array.ndim == 0:
+        if data_array.notnull().item():
+            return data_array.to_numpy()[()]
+        return np.nan
+
+    flat = data_array.stack(_flat=data_array.dims)
+    valid = flat.where(flat.notnull(), drop=True)
+    if valid.size:
+        return valid.isel(_flat=0).to_numpy()[()]
     return np.nan
 
 

--- a/xradar/transform/cfradial.py
+++ b/xradar/transform/cfradial.py
@@ -66,6 +66,7 @@ from ..io.backends.cfradial1 import (
     _get_required_root_dataset,
     _get_subgroup,
     _get_sweep_groups,
+    open_cfradial1_datatree,
 )
 from ..io.backends.common import _attach_sweep_groups
 from ..io.export.cfradial1 import (
@@ -133,24 +134,46 @@ def to_cfradial2(ds, **kwargs):
     kwargs.pop("site_as_coords", True)
     sweep = kwargs.pop("sweep", None)
 
+    # Datasets opened via ``engine="cfradial1"`` without a group only expose the
+    # normalized root metadata view. If the original source path is available,
+    # reopen the full volume and delegate to the standard datatree reader.
+    if "sweep_number" not in ds and "fixed_angle" not in ds:
+        if "sweep_group_name" in ds and "sweep_fixed_angle" in ds:
+            source = ds.encoding.get("source")
+            if source:
+                return open_cfradial1_datatree(
+                    source,
+                    sweep=sweep,
+                    first_dim=first_dim,
+                    optional=optional,
+                    optional_groups=True,
+                )
+            raise ValueError(
+                "to_cfradial2 requires a full CfRadial1 volume dataset containing "
+                "`sweep_number` and `fixed_angle`. "
+                "The provided dataset looks like the normalized root metadata view "
+                'returned by `xarray.open_dataset(..., engine="cfradial1")`, but '
+                "it does not retain a readable `encoding['source']` path to reopen "
+                "the full volume."
+            )
+
     # Create DataTree root node with required data
     root_data = _get_required_root_dataset(ds, optional=optional)
     dtree = DataTree(dataset=root_data, name="root")
 
     # Attach additional root metadata groups as child nodes
     radar_parameters = _get_subgroup(ds, radar_parameters_subgroup)
-    if radar_parameters:
+    if radar_parameters.data_vars:
         dtree["radar_parameters"] = DataTree(radar_parameters, name="radar_parameters")
 
     calib = _get_radar_calibration(ds)
-    if calib:
+    if calib.data_vars:
         dtree["radar_calibration"] = DataTree(calib, name="radar_calibration")
 
     georeferencing = _get_subgroup(ds, georeferencing_correction_subgroup)
-    if georeferencing:
-        dtree["georeferencing_correction"] = DataTree(
-            georeferencing, name="georeferencing_correction"
-        )
+    dtree["georeferencing_correction"] = DataTree(
+        georeferencing, name="georeferencing_correction"
+    )
 
     # Attach sweep child nodes
     sweep_groups = list(

